File: kotlin/ktsearch/src/main/kotlin/ktsearch/SearchFile.kt
package ktsearch

import java.io.File

/**
 * @author cary on 7/24/16.
 */
class SearchFile(val containers: List<String>, val file: File, val fileType: FileType) {
    val CONTAINER_SEPARATOR = "!"

    constructor(file: File, fileType: FileType) : this(listOf(), file, fileType)

    override fun toString(): String {
        val sb = StringBuilder()
        if (containers.isNotEmpty()) {
            for (i in containers.indices) {
                if (i > 0) {
                    sb.append(CONTAINER_SEPARATOR)
                }
                sb.append(containers[i])
            }
            sb.append(CONTAINER_SEPARATOR)
        }
        sb.append(file.path)
        return sb.toString()
    }
}

File: dart/dartsearch/lib/src/search_file.dart
import 'dart:io' show File;

import 'package:dartsearch/src/file_types.dart';

class SearchFile {
  static const String CONTAINER_SEPARATOR = '!';

  List<String> containers = [];
  final File file;
  final FileType fileType;

  SearchFile(this.file, this.fileType);

  @override
  String toString() {
    var s ='';
    if (containers.isNotEmpty) {
      s = containers.join(CONTAINER_SEPARATOR) + CONTAINER_SEPARATOR;
    }
    s += file.path;
    return s;
  }
}
